WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -1,15 +1,67 @@
 package org.wordpress.android.ui.jetpack.scan
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanIdleState
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanScanningState
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
 import javax.inject.Inject
 
-class ScanViewModel @Inject constructor() : ViewModel() {
+class ScanViewModel @Inject constructor(
+    private val scanStatusService: ScanStatusService
+) : ViewModel() {
     private var isStarted = false
+    private var lastScanState: ScanStateModel.State? = null
```

Comment:
💡 Idea / 🔍 Nitpick: I'm wondering if it'd be cleaner to always access `scanStatusService.scanState` instead of introducing a variable. Wdyt?

Reply:
Seems reasonable to me.

Reply:
It came by referring to a similar check in the `ActivityLogModel` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/viewmodel/activitylog/ActivityLogViewModel.kt#L125) showing  a rewind progress indicator. It might have become useful when we would've started showing a scan progress indicator. However, I'm removing it for now and re-add if needed.

In ca0c82a